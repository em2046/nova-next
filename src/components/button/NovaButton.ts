import { defineComponent, h, VNode } from 'vue'

export default defineComponent({
  setup(props, context) {
    const { slots } = context

    return (): VNode => {
      const children = slots.default && slots.default()
      return h(
        'button',
        {
          class: 'nova-button',
          type: 'button',
        },
        children
      )
    }
  },
})
