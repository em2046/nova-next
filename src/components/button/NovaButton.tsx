import { defineComponent } from 'vue';
import { vueJsxCompat } from '../../vue-jsx-compat';
import useEnvironment, {
  EnvironmentProps,
  environmentProps,
} from '../../uses/useEnvironment';

export default defineComponent({
  name: 'NovaButton',
  props: {
    ...environmentProps,
  },
  setup(props, context) {
    const { slots } = context;

    const environment = useEnvironment((props as unknown) as EnvironmentProps);

    return (): JSX.Element => {
      const children = slots.default?.();
      return (
        <button
          class="nova-button"
          type="button"
          data-nova-theme={environment.themeRef.value}
        >
          {children}
        </button>
      );
    };
  },
});
