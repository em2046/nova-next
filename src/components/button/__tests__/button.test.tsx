import { mount } from '@vue/test-utils';
import { vueJsxCompat } from '../../../vue-jsx-compat';
import { NovaButton } from '../NovaButton';

describe('button', () => {
  test('render', () => {
    const wrapper = mount({
      setup() {
        return () => {
          return (
            <div>
              <NovaButton>{() => 'Button'}</NovaButton>
              <NovaButton primary>{() => 'Primary'}</NovaButton>
              <NovaButton disabled>{() => 'Disabled'}</NovaButton>
            </div>
          );
        };
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
