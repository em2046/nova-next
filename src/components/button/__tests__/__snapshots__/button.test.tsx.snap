// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button render 1`] = `"<div><button class=\\"nova-button\\" type=\\"button\\" data-nova-theme=\\"light\\"><!----><span>Button</span></button><button class=\\"nova-button nova-button-primary\\" type=\\"button\\" data-nova-theme=\\"light\\"><!----><span>Primary</span></button><button class=\\"nova-button\\" type=\\"button\\" data-nova-theme=\\"light\\" disabled=\\"\\"><!----><span>Disabled</span></button></div>"`;
